File: src/main/java/com/vspk/bookrest/security/JwtTokenProvider.java
package com.vspk.bookrest.security;

import com.vspk.bookrest.domain.Role;
import org.springframework.security.core.Authentication;

import javax.servlet.http.HttpServletRequest;
import java.util.List;

public interface JwtTokenProvider {
    String createToken(String username, List<Role> roles);

    Authentication getAuthentication(String token);

    String getUsername(String token);

    String resolveToken(HttpServletRequest req);

    boolean isTokenValid(String token);
}

File: src/main/java/com/vspk/bookrest/security/JwtTokenProvider.kt
package com.vspk.bookrest.security

import com.vspk.bookrest.domain.Role
import javax.servlet.http.HttpServletRequest
import org.springframework.security.core.Authentication

interface JwtTokenProvider {
    fun createToken(username: String, roles: List<Role>): String
    fun getAuthentication(token: String): Authentication
    fun getUsername(token: String): String
    fun resolveToken(req: HttpServletRequest): String
    fun isTokenValid(token: String): Boolean
}